### PR TITLE
Removed some irrelevant questions

### PR DESCRIPTION
**Pull request by mluisbrown**
And cleaned up some others.

### CODE REVIEW

File: Interview/questions.md [hunk 1]
```diff
@@ -105,19 +93,15 @@ Please keep in mind that this document is in a continuous state of flux, with ne
...
 - What is a `thread`? In which cases would you create your own threads?
-- What is a `runloop`? What is a `runloop mode`?
 - What do you think of `Grand Central Dispatch`?
 - Have you used `dispatch_once`? In broad strokes, how does it work?
-- Have you used `dispatch_barrier`? When is it appropriate and useful to use?
-- What is a `dispatch_group`?
-- Have you used `NSOperationQueue` and `NSOperation`?
+- What is a `dispatch group`?
 - What methods for thread synchronisation do you know on iOS?
-- What are the main differences between `GCD` queues and `NSOperationQueue`s?
```
Comment (RuiAAPeres) [inline Interview/questions.md]: >  Have you used `NSOperationQueue` and `NSOperation`?
>  What are the main differences between `GCD` queues and `NSOperationQueue`s?

Are IMO relevant questions to be asked.

File: Interview/questions.md [hunk 2]
```diff
@@ -75,7 +63,7 @@ Please keep in mind that this document is in a continuous state of flux, with ne
 - Is it possible to define an instance method that applies to arrays of integers but *not* to arrays of doubles, or any other type? Can you think of an example when this could be useful? Do you happen to know the syntax?
 - What is the Swift embedded runtime?
 - What is the difference between `value semantics` and `reference semantics`?
-- So, say that you have a struct and one or more of its properties is of a reference type. What happens then? Is the struct still a value type?
+- So, say that you have a struct and one or more of its properties is of a reference type. What happens then? Does the struct still have value semantics?
```
Comment (diegopetrucci) [inline Interview/questions.md]:
```suggestion
- Say that you have a struct and one or more of its properties is of a reference type. What happens then? Does the struct still have value semantics?
```

File: Interview/questions.md [hunk 3]
```diff
@@ -75,7 +63,7 @@ Please keep in mind that this document is in a continuous state of flux, with ne
...
 - How do the Obj-C and Swift runtime environments manage memory these days?
```
Comment (diegopetrucci) [inline Interview/questions.md]:
```suggestion
- How does the Swift runtime environment manage memory these days?
```

File: Interview/questions.md [hunk 4]
```diff
@@ -105,19 +93,15 @@ Please keep in mind that this document is in a continuous state of flux, with ne
...
 - What do you think of `Grand Central Dispatch`?
 - Have you used `dispatch_once`? In broad strokes, how does it work?
-- Have you used `dispatch_barrier`? When is it appropriate and useful to use?
-- What is a `dispatch_group`?
-- Have you used `NSOperationQueue` and `NSOperation`?
+- What is a `dispatch group`?
 - What methods for thread synchronisation do you know on iOS?
-- What are the main differences between `GCD` queues and `NSOperationQueue`s?
 - What are the advantages and disadvantages of `GCD` and `NSOperationQueue`/`NSOperations`?
 - What is `KVO`? Do you know how it works?
```
Comment (diegopetrucci) [inline Interview/questions.md]: Do we need this?

File: Interview/questions.md (hunk 1 above)
Comment (mluisbrown) [reply Interview/questions.md]: Do we really need 3 different questions comparing GCD and NSOperation / NSOperationQueue? (see question immediately below)  Also, IMO questions like `Have you used X?` without a followup are meaningless. Both Yes or No are correct answers.

File: Interview/questions.md (hunk 4 above)
Comment (mluisbrown) [reply Interview/questions.md]: IMO it's more important to know about KVO than NSOperations...

**Pull request merged**